### DISCUSSION TITLE
Usage guide?

### DISCUSSION THREAD

**Discussion by sengokudaikon** [huanshankeji/exposed-vertx-sql-client#7] [Q&A]
Hi, i want to try to use this library because i like Exposed and Vert.x client, can you provide at least a basic usage guide on how to use this? Thanks in advance

**Comment by sengokudaikon** [comment]
Most importantly, i would like to know how to execute inserts and updates, the library isn't very clear on how to use DataMapper or reflectionBasedClassPropertyDataMapper. when trying to import it as is, it's not recognized by the compiler as a valid import

**Reply by ShreckYe** [reply]
> i would like to know how to execute inserts and updates

Please see the newly added usage guide.

> the library isn't very clear on how to use DataMapper or reflectionBasedClassPropertyDataMapper

`reflectionBasedClassPropertyDataMapper` is a very experimental feature implemented in a separate library [Exposed GADT Mapping](https://github.com/huanshankeji/exposed-adt-mapping) and I will provide a basic usage guide and a new release 0.1.1 or 0.2.0 soon. `DataMapper` is currently only a super interface for `ReflectionBasedClassPropertyDataMapper`, and if you don't use `ReflectionBasedClassPropertyDataMapper` now you most likely don't need `DataMapper`.

> when trying to import it as is, it's not recognized by the compiler as a valid import

This is indeed an IntelliJ IDEA problem, I am planning to fix it in #6.

**Comment by ShreckYe** [comment]
Hi, thank you very much for your interest. I am very sorry for not replying in time. It seems that because I didn't watch this project I didn't get any notifications for this question.

I have updated the README with a basic usage guide. There are still some old APIs that will be refactored. Hopefully, I will fix them by the end of October in a new release 0.4.0 or 0.5.0.
